### PR TITLE
Switch for disabling fix text encoding

### PR DESCRIPTION
**Pull request by ccottingham-presinet**
Olaf:
Hi.
This is my first pull request pn github so I hope I did ok.

Please feel free to change the wording etc

Thanks,
Clay

### CODE REVIEW

File: lib/HTML/Restrict.pm [hunk 1]
```diff
@@ -71,6 +71,12 @@ has trim => (
     default => 1,
 );
 
+has process_text => (
```
Comment (oalders) [inline lib/HTML/Restrict.pm:74]: This was my first thought as well, but I think we can tweak this slightly to make the API even more flexible. How about we follow the pattern of `replace_img`?

```perl
has replace_img => (
    is      => 'rw',
    isa     => Bool | CodeRef,
    default => 0,
);
```

I think the name of the attribute you've chosen makes sense.

If we follow this pattern we can either process the text as is currently the default, do nothing or allow the user to provide their own processing sub. That feels like a pretty flexible solution.

**Review by oalders — CHANGES_REQUESTED** [review]
Thank you so much for this. It's a great first GitHub Pull Request. 😄

**Comment by oalders** [comment]
@haarg any thoughts on this?

**Comment by oalders** [comment]
I just added GitHub workflows for this repo, so if you rebase we'll get testing on this branch as well.

File: README.md [hunk 2]
```diff
@@ -283,6 +283,11 @@ HTML::Restrict recognizes:
     enclosed content you'd like to remove.  This feature strips script and style
     tags by default.
 
+- `process_text => [0|1]`
+
+    By default all text will be processed.  Set this value to 0 in order to disable this behaviour.  Please be advised this is a security risk. Use caution when disabling this parameter.
+
+
```
Comment (haarg) [inline README.md:290]: These docs are in the wrong location. The `README.md` file is a generated file, based on the docs in the `lib/HTML/Restrict.pm` file.

I also don't find them very clear. What does "processed" mean in this context?

Comment (oalders) [reply README.md:290]: Maybe `filter_text()` would be clearer?

Comment (ccottingham-presinet) [reply README.md:290]: @haarg 
I didnt realize they were generated from module I can make that change easily enough.

File: lib/HTML/Restrict.pm (hunk 1 above)
Comment (ccottingham-presinet) [reply lib/HTML/Restrict.pm:74]: Sounds good, 
Ill get that in ASAP

File: README.md (hunk 2 above)
Comment (ccottingham-presinet) [reply README.md:290]: Let me know how you want this to be named. Variable naming is usually the hardest part of coding!

We have a release starting today so I will try to get these in asap

Comment (oalders) [reply README.md:290]: I think `filter_text()` would work. If there are little things like naming, I can just update the commits before merge. That's no problem at all. Thanks for all of your help with this.

**Comment by ccottingham-presinet** [comment]
I just pushed a new commit. 
Please take a look at your convenience.

**Comment by oalders** [comment]
Thanks very much for this @ccottingham-presinet! I've merged it at the command line as it just needed a few tweaks like code tidying and some doc changes. Your contribution is very much appreciated. I'll upload a new release shortly.

**Comment by ccottingham-presinet** [comment]
thanks for accepting this commit. and making my first PR so easy!
have a good weekend

**Pull request closed without merging**